EIPS/eip-4881.md

```diff
@@ -20,15 +20,15 @@ Most client implementations require beacon nodes to download and store every dep
 Consensus clients MAY continue to implement the deposit Merkle tree however they choose. However, when transmitting the tree to newly syncing nodes, clients MUST use the following format:
 
 ```
-DepositTreeSnapshot {
-    finalized: Vector[Hash32],
-    deposits: uint64,
-    execution_block_hash: Hash32,
-}
+class DepositTreeSnapshot:
+    finalized: List[Hash32, DEPOSIT_CONTRACT_DEPTH]
+    deposit_root: Hash32
+    deposit_count: uint64
+    execution_block_hash: Hash32
+    execution_block_height: uint64
 ```
 
-Where the hashes in the `finalized` vector are defined in the [Deposit Finalization Flow](#deposit-finalization-flow) section below, `deposits` is the number of deposits stored in the snapshot, and `execution_block_hash` is the hash of the execution block containing the highest index deposit stored in the snapshot. Consensus clients MUST make this structure available via the Beacon Node API endpoint:
-
+Where `finalized` is a variable-length list (of maximum size `DEPOSIT_CONTRACT_DEPTH`) containing the hashes defined in the [Deposit Finalization Flow](#deposit-finalization-flow) section below. The fields `deposit_root`, `deposit_count`, and `execution_block_hash` store the same information as the [`Eth1Data`](https://github.com/ethereum/consensus-specs/blob/v1.1.9/specs/phase0/beacon-chain.md#eth1data) object that corresponds to the snapshot, and `execution_block_height` is the height of the execution block with hash `execution_block_hash`. Consensus clients MUST make this structure available via the Beacon Node API endpoint:
```

Comment:
I think this link shall be allowed.

Reply:
CC @SamWilsn

I am mildly against including this at this time, as this contradicts EIP-1. @xinbenlv if you would like to add a change to EIP-1 allowing links to commits of the consensus specs in a new PR, go ahead, as it will likely prompt discussion.

Reply:
@Pandapip1 
There are 213 other EIPs that link to `https://github.com/ethereum`, one of which is this one as it currently exists. This PR doesn't modify those links at all.

Is it really necessary that my update to this EIP, which is needed for people who are currently implementing it, should wait for this discussion to play out?

Reply:
Yes, it is absolutely necessary. Those EIPs were made when the external links policy was more relaxed.

Sorry for the hassle. The link does need to be removed, though.

Reply:
EIP-1 states that links to external resources **SHOULD NOT** be included. Therefore it is not "absolutely necessary" for the links to be removed. And as I said above, I would like `ethereum/consensus-specs` added to the allow list. It makes no sense to not allow EIP authors to link to `ethereum/consensus-specs`. The beacon chain _is_ Ethereum and we shouldn't be asking authors to needlessly duplicate spec code in their EIP.

Reply:
<img width="769" alt="image" src="https://user-images.githubusercontent.com/29681685/185115492-94a6de68-3075-4ec7-9a6d-de05c8c1c763.png">

Looking into errors, it appears to me that EIPW checked for non-relative links and thus found error with specs link and missed ref of EIP-4444 (because it is added as relative link in the proposal). 

<img width="105" alt="image" src="https://user-images.githubusercontent.com/29681685/185114369-339c5257-288b-4016-8727-01b8b8659e6d.png">

IMO, specs link are not "external" as they are part of the same GitHub organization. Probably "external resources" could be defined in EIP-1.

Reply:
I am okay with letting this one link slide for now.

Reply:
> And as I said above, I would like ethereum/consensus-specs added to the allow list. It makes no sense to not allow EIP authors to link to ethereum/consensus-specs. The beacon chain is Ethereum and we shouldn't be asking authors to needlessly duplicate spec code in their EIP.

+1 to this.